frontend/src/components/UserAuth/UserSignUp.tsx

```diff
@@ -1,43 +1,148 @@
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Email field max character up to 255 only.',
+    },
+  },
+  password: {
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Password field max character up to 255 only.',
+    },
+    minLength: {
+      value: 6,
+      message: 'Password field min character is 6.',
+    },
+  },
+  password_confirmation: {
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Password field max character up to 255 only.',
+    },
+    minLength: {
+      value: 6,
+      message: 'Password field min character is 6.',
+    },
+  },
+};
 
 export const UserSignUp = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormInput>();
+
+  const navigate = useNavigate();
+  const [isError, setIsError] = useState(false);
+  const [cookies, setCookies] = useCookies();
+
+  const onSubmit = (data: FormInput) => {
+    setIsError(false);
+    backend.get('/sanctum/csrf-cookie').then((csrf_response) => {
+      backend
+        .post('/api/users/', data)
+        .then((response: AxiosResponse) => {
+          setCookies('user', response.data.data);
+          setCookies('token', response.data.token);
+          navigate('/signin');
+        })
+        .catch((err: AxiosError) => {
+          setIsError(true);
+        });
+    });
+  };
+
   return (
     <div className="min-h-full flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
       <div className="max-w-md w-full space-y-8">
         <UserAuthHeader>Register to Stephen's E-Learning System</UserAuthHeader>
 
-        <UserAuthForm action="#" method="POST">
+        <UserAuthForm handleSubmit={handleSubmit(onSubmit)}>
           <div className="rounded-md shadow-sm -space-y-px">
             <UserAuthField
               label="Full Name"
-              name="name"
+              register={{ ...register('name', formValidation.name) }}
               rounded="t-md rounded-b-md"
             />
+            <div>{errors.password?.message}</div>
           </div>
 
           <div className="rounded-md shadow-sm -space-y-px">
             <UserAuthField
               label="Email Address"
-              name="email"
+              register={{ ...register('email', formValidation.email) }}
               type="email"
               rounded="t-md"
             />
-            <UserAuthField label="Password" name="password" rounded="none" />
+            <div>{errors.password?.message}</div>
+            <UserAuthField
+              label="Password"
+              register={{ ...register('password', formValidation.password) }}
+              type="password"
+              rounded="none"
+            />
+            <div>{errors.password?.message}</div>
```

Comment:
since you are creating a component for userAuthfield, you can also pass the error message.

Reply:
had some changed